core/iwasm/interpreter/wasm_loader.c

```diff
@@ -830,32 +830,49 @@ load_init_expr(WASMModule *module, const uint8 **p_buf, const uint8 *buf_end,
             case INIT_EXPR_TYPE_REFNULL_CONST:
             {
                 uint8 type1;
-
-#if WASM_ENABLE_GC == 0
+#if WASM_ENABLE_GC != 0
+                const uint8 *p_copy = p;
+                int32 heap_type;
+                read_leb_int32(p_copy, p_end, heap_type);
+#endif
```

Comment:
If following the spec, s33 should be a type index. might rename it as type_index better.

Reply:
I think it might not be appropriate to rename `heap_type` to `type_idx` here:
1. `type_idx` has already been defined as a `uint32` at the beginning of the function.
2. Based on the surrounding lines of code in function `wasm_loader_prepare_bytecode` when `WASM_OP_REF_NULL`, I think it's clearer and more consistent to keep the name heap_type.
https://github.com/bytecodealliance/wasm-micro-runtime/blob/18d4227ab6593bb1870aac025d351cce9e3c1041/core/iwasm/interpreter/wasm_loader.c#L13154-L13169